interfaces/remoteOperatorUI/ffmpegInterface.js

```diff
@@ -0,0 +1,191 @@
+            // '-pix_fmt', 'yuv420p',
+            '-pix_fmt', 'argb',
+            // '-vf', 'scale=' + outputWidth + ':' + outputHeight + ', setsar=1:1', //, realtime, fps=' + framerate,
+            // '-preset', 'ultrafast',
+            // '-copyts',
+            // '-tune', 'zerolatency',
+            // '-r', framerate, // will duplicate frames to meet this but still look like the framerate set before -i,
+            output_path
+        ];
+
+        // not working with MP4 ... works losslessly with .MKV but not playable in HTML video element
+        // let args = [
+        //     '-r', framerate,
+        //     '-f', 'image2pipe',
+        //     '-vcodec', input_vcodec,
+        //     '-s', input_width + 'x' + input_height,
+        //     '-i', '-',
+        //     '-vcodec', 'libx265',
+        //     '-x265-params', 'crf=0:lossless=1:preset=veryslow:qp=0',
+        //     // '-pix_fmt', 'yuv420p',
+        //     // '-vf', 'scale=' + outputWidth + ':' + outputHeight + ', setsar=1:1', //, realtime, fps=' + framerate,
+        //     // '-preset', 'ultrafast',
+        //     // '-copyts',
+        //     // '-tune', 'zerolatency',
+        //     // '-r', framerate, // will duplicate frames to meet this but still look like the framerate set before -i,
+        //     output_path
+        // ];
+
+        let process = cp.spawn(ffmpegPath, args);
+
+        if (constants.DEBUG_LOG_FFMPEG) {
+            // process.stdout.on('data', function(data) {
+            //     console.log('stdout data', data);
+            // });
+            process.stderr.setEncoding('utf8');
+            process.stderr.on('data', function (data) {
+                console.log('stderr data', data);
+            });
+            // process.on('close', function() {
+            //     console.log('finished');
+            // });
+        }
+
+        return process;
+    },
+    // takes in a textfile with a list of filepaths, one per line, e.g:
+    //      file '/my/file/path1.mp4'
+    //      file '/my/file/path2.mp4'
+    // and concatenates the specified video segments into a single video
+    ffmpeg_concat_mp4s: (output_path, file_list_path) => {
+        // ffmpeg -f concat -safe 0 -i fileList.txt -c copy mergedVideo.mp4
+        // we pass in a timestamp so we can use an identical one in the color and depth videos that match up
+        let args = [
+            '-f', 'concat',
+            '-safe', '0',
+            '-i', file_list_path,
+            '-c', 'copy',
+            output_path
+        ];
+
+        let process = cp.spawn(ffmpegPath, args);
+        return process;
+    },
+    // takes a video and resamples the timestamps of each frame to make it last the specified duration (in seconds)
+    // currently relying on the VideoLib.MovieParser to get the length of the video file, which is a bit flaky,
+    // as such this method isn't currently used and the VideoLib is commented out to avoid installing that dependency
```

Comment:
It looks like you can get the video duration using ffmpeg or ffprobe (which is usually downloaded alongside ffmpeg): https://superuser.com/questions/650291/how-to-get-video-duration-in-seconds

Reply:
Thanks for the tip! Seems to work, I'll push an update